chain/store/messages.go

```diff
@@ -300,3 +301,144 @@ func (cs *ChainStore) LoadSignedMessagesFromCids(ctx context.Context, cids []cid
 
 	return msgs, nil
 }
+
+// TipSetBlockMessagesReceipts returns the blocks and messages in `ts` and their corresponding receipts from `pts` matching block order in tipset (`ts`).
+func (cs *ChainStore) TipSetBlockMessagesReceipts(ctx context.Context, ts, pts *types.TipSet) ([]*BlockMessageReceipts, error) {
+	// returned BlockMessages match block order in tipset
```

Comment:
I'd sanity check input here: both ts and pts should be non-empty, and pts should be the parent of ts

Reply:
done via [a07ee7e](https://github.com/filecoin-project/lotus/pull/9186/commits/a07ee7e6ef3d5b124efcf5213b76f598649c6281)